osu.Game.Rulesets.Osu/Mods/OsuModDifficultyAdjust.cs

```diff
@@ -53,5 +58,31 @@ protected override void ApplySettings(BeatmapDifficulty difficulty)
             if (CircleSize.Value != null) difficulty.CircleSize = CircleSize.Value.Value;
             if (ApproachRate.Value != null) difficulty.ApproachRate = ApproachRate.Value.Value;
         }
+
+        private partial class ApproachRateSettingsControl : DifficultyAdjustSettingsControl
+        {
+            protected override Drawable CreateControl() => new SliderControl(SliderDisplayCurrent,
+                new ApproachRateSlider
+                {
+                    RelativeSizeAxes = Axes.X,
+                    Current = SliderDisplayCurrent,
+                    KeyboardStep = 0.1f,
+                }
+            );
+
+            /// <summary>
+            /// A slider bar with more detailed approach rate info for its given value
+            /// </summary>
+            public partial class ApproachRateSlider : RoundedSliderBar<float>
+            {
+                public override LocalisableString TooltipText =>
+                    $"{base.TooltipText} ({millisecondsFromApproachRate(Current.Value, 1.0f)} ms)";
+
+                private double millisecondsFromApproachRate(float value, float clockRate)
+                {
+                    return Math.Round(1800 - Math.Min(value, 5) * 120 - (value >= 5 ? (value - 5) * 150 : 0) / clockRate);
```

Comment:
I don't know where you got this from, but it doesn't match any existing implementations and is probably wrong. I'll fix it.

Reply:
Fixed by just creating a hitcircle and reading the raw value out of it.

I also nuked the `clockRate` param (I get what this was going for, but we're not handling "effective" AR anywhere yet, so this should be done separately and comprehensively), and made the ms thing display only when extended limits are active. That last one is kinda style, kinda not wanting to overwhelm users / prompt questions "why does only osu! ruleset get this ms display".